File: app/local_plugins/core/ai_plugin_interface/android/src/main/kotlin/com/aiagent/plugin_interface/NativeTranslationService.kt
package com.aiagent.plugin_interface

/**
 * 翻译服务原生接口
 *
 * 实现方：translation_deepl, translation_aliyun 等服务插件
 * 使用方：agent_translate
 *
 * 职责：文本翻译（HTTP API 调用）
 */
interface NativeTranslationService {

    /**
     * 初始化翻译服务
     * @param configJson  服务配置 JSON（apiKey 等）
     */
    fun initialize(configJson: String)

    /**
     * 翻译文本
     *
     * @param text          要翻译的文本
     * @param targetLang    目标语言代码（如 "en", "zh"）
     * @param sourceLang    源语言代码（可选，null 表示自动检测）
     * @return 翻译结果
     */
    suspend fun translate(
        text: String,
        targetLang: String,
        sourceLang: String? = null,
    ): NativeTranslationResult

    /** 释放资源 */
    fun release()
}

File: app/local_plugins/core/ai_plugin_interface/ios/Classes/NativeTranslationService.swift
import Foundation

/// Text-translation service contract.
///
/// Pure request/response — no events, no requestId, no interruption.
public protocol NativeTranslationService: AnyObject {
    func initialize(configJson: String)

    /// Translate `text` into `targetLang`. `sourceLang == nil` → auto-detect.
    func translate(text: String, targetLang: String, sourceLang: String?) async throws
        -> NativeTranslationResult

    func release()
}

public extension NativeTranslationService {
    func translate(text: String, targetLang: String) async throws -> NativeTranslationResult {
        try await translate(text: text, targetLang: targetLang, sourceLang: nil)
    }
}

public struct TranslationException: Error {
    public let code: String
    public let message: String
    public init(code: String, message: String) {
        self.code = code
        self.message = message
    }
}
